### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Yerimah**
In this branch we worked on the following:
- Created the remaining classes
- Created associations between them
- Fix linter errors

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require './rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rental
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
```
Comment (Deepakdanger) [inline book.rb]: - [ ] I would recommend keeping the `rental` variable plural(`rentals`) as book has the relation `has many` with rentals.
- [ ] I would recommend adding both Getters and Setters for `@rentals`

Hint:
`attr_accessor :rentals

File: person.rb [hunk 2]
```diff
@@ -9,6 +10,7 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
```
Comment (Deepakdanger) [inline person.rb]: - [ ] I would recommend keeping the `rental` variable plural (`rentals`) as book has the relation `has many` with rentals.
- [ ] I would recommend adding both Getters and Setters for `@rentals`

Hint:
`attr_accessor :rentals

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+class Rental
+  attr_accessor :person, :book
+  attr_reader :date
+
+  def initialize(person, date, book)
+    @date = date
+    @person = person
+    person.rental.push(self)
+    @book = book
+    book.rental.push(self)
```
Comment (Deepakdanger) [inline rental.rb]: - [ ] After changing the variable name `rentals` make changes here as well.

Hint:
```ruby
person.rentals.push(self)
   ...
book.rentals.push(self)
```

File: classroom.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Classroom
+  attr_accessor :label
+  attr_reader :student
+
+  def initialize(label)
+    @label = label
+    @student = []
```
Comment (Deepakdanger) [inline classroom.rb]: - [ ] I would recommend keeping the `student` variable plural (`students`) as classroom has the relation `has many` with students.

File: student.rb [hunk 5]
```diff
@@ -9,4 +11,9 @@ def initialize(classroom, age, name = 'unknown', parent_permission: true)
   def play_hooky
     "¯\(ツ)/¯"
   end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.student.push(self) unless classroom.student.include?(self)
```
Comment (Deepakdanger) [inline student.rb]: - [ ] After changing the variable name `students` make changes here as well.

Hint:
```ruby
classroom.students.push(self) unless classroom.students.include?(self)
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Yerimah, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight :100: 
:+1:  PR description looks good.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @Yerimah, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Yerimah** [comment]
THANK YOU

**Pull request merged**